selfdrive/modeld/models/dmonitoring.cc

```diff
@@ -114,24 +114,39 @@ DMonitoringResult dmonitoring_eval_frame(DMonitoringModelState* s, void* stream_
                     resized_width, resized_height,
                     mode);
 
+  // prerotate to be cache aware
+  uint8_t *resized_buf_rot = get_buffer(s->resized_buf_rot, resized_width*resized_height*3/2);
+  uint8_t *resized_y_buf_rot = resized_buf_rot;
+  uint8_t *resized_u_buf_rot = resized_y_buf_rot + (resized_width * resized_height);
+  uint8_t *resized_v_buf_rot = resized_u_buf_rot + ((resized_width/2) * (resized_height/2));
+
+  libyuv::I420Rotate(resized_y_buf, resized_width,
+                     resized_u_buf, resized_width/2,
+                     resized_v_buf, resized_width/2,
+                     resized_y_buf_rot, resized_height,
+                     resized_u_buf_rot, resized_height/2,
+                     resized_v_buf_rot, resized_height/2,
+                     // negative height causes a vertical flip to match previous
+                     resized_width, -resized_height, libyuv::kRotate90);
+
   int yuv_buf_len = (MODEL_WIDTH/2) * (MODEL_HEIGHT/2) * 6; // Y|u|v -> y|y|y|y|u|v
   float *net_input_buf = get_buffer(s->net_input_buf, yuv_buf_len);
   // one shot conversion, O(n) anyway
   // yuvframe2tensor, normalize
   for (int c = 0; c < MODEL_WIDTH/2; c++) {
     for (int r = 0; r < MODEL_HEIGHT/2; r++) {
       // Y_ul
-      net_input_buf[(c*MODEL_HEIGHT/2) + r] = input_lambda(resized_buf[(2*r*resized_width) + (2*c)]);
-      // Y_ur
-      net_input_buf[(c*MODEL_HEIGHT/2) + r + (2*(MODEL_WIDTH/2)*(MODEL_HEIGHT/2))] = input_lambda(resized_buf[(2*r*resized_width) + (2*c+1)]);
+      net_input_buf[(c*MODEL_HEIGHT/2) + r + (0*(MODEL_WIDTH/2)*(MODEL_HEIGHT/2))] = input_lambda(resized_buf_rot[(2*r) + (2*c)*resized_height]);
       // Y_dl
```

Comment:
ur and dl should be swapped back since it's rotated.
besides that everything lgtm.